### ISSUE TITLE
Consider license Anytype under a open source license

### ISSUE THREAD

**Issue by danny-laa** [anyproto/anytype-ts#79]
## Description 

It is advertised that [Anytype is open source software](https://anytype.io/), but the [Any Source Available License](https://github.com/anyproto/anytype-ts/blob/main/LICENSE.md) doesn't seem to meet the common expectation of "Open Source".

It would be better if Anytype is licensed in either,

- AGPL, which makes it hard to commercialize the software
- Business Source License, which forbids production use, but makes a particular version of the software open source several years after that version is released.

**Comment by louwers** [comment]
The Business Source License is not an "open source" license either.

I don't understand why the server backend is released under MIT but the client code is released under a non-free license. If you are afraid others will sell the client code you can use a GPL license and everyone who distributes the software needs to do so under a free license as well.

**Comment by fuksman** [comment]
@danny-laa, sorry for the delayed response, there were a lot of activities after the launch.

@danny-laa, @louwers  
We'll fix our marketing messages to better reflect the nuances of licensing: protocols and infrastructure layer are open-source (MIT), apps are source-available.

We have published the reasoning behind this approach in the article: https://blog.anytype.io/our-open-philosophy/

BSL is not an open-source license either, but we do consider changing the license in the future. We did not include a "timer" in the license text because there are examples of projects which regularly move this timer further into the future due to business reasons. In our opinion, it is better to avoid such situations.

AGPL or GPL could be "uncomfortable", but still, as open source licenses, they don't have any restrictions on reselling the product, hence, they don't protect from rivals' abuse.

When we thought about implementing a license that would protect the team and contributors from competitive commercial use, we had several possible options (of course, we didn't take keeping the source code closed, as it would contradict our promise to users): 
1. Source-available license for everything: great level of protection from abuse, but there are risks for user data.
2. Source-available infrastructure and protocols, open-source clients: good level of protection from abuse, unrestricted use of app code, but there are risks for user data.
3. Open-source infrastructure and protocols, source-available clients: good level of protection from abuse and no risks for user data. The license for apps created by us has restrictions for competitive commercial use, but everyone is and will be able to develop alternative clients relying on open-source protocols and infrastructure.

So, thinking about all these options, we see that the best option is open-sourcing infrastructure and protocols, and providing limited restrictions for clients' code in terms of competitive commercial use. And we did it, as it answers the demands of both users and creators.

This and other popular questions are answered on https://legal.any.coop/faq, and we're happy to address more questions.

I hope it answers all raised questions and close this issue as not-to-be-implemented in the scope of `anytype-ts`. It doesn't mean that the discussion is over: we would like to answer any licensing questions in the contributor's community ([this](https://github.com/orgs/anyproto/discussions/1) can be the best place) or via [email](mailto:license@any.coop).

**Comment by flypenguin** [comment]
I thought I remembered you being a "real" open source company, and that "source available" mention on the web page made me stop. I was pretty surprised that you weren't, yet I was really really sure. So I went and had a look.

Turns out, you changed the license _from_ MIT (commit e109b5556) _to_ Source Available (commit deb782c83) in May 2023, which is fairly recently.

**That you were open source was _really_ important for me**. I really don't want to get stuck with unrecoverable data if any (let's hope not, but I plan to use software for more than a couple of years – and a lot can happen in there) goes belly up – or, more realistic, gets bought out. And we have all seen what usually happens with the product _then_

The only "defense" against that is open source.

I would love you to switch to AGPL (why would _anyone_ buy the software from someone else, if the code is public, and you folks are "the real deal"??) – as, for example, [appflowy](https://www.appflowy.io/) does. I would also happily pay for the product, still. I know people need to make money. Now? Not so much, I'd rather get the money into a "migration fonds", expecting the worst.

Because let's be honest – if the client goes "away" for whatever reason: do you _honestly_ expect anyone to re-write it on the basis of your "open protocols", just to continue using the software? Even if somebody does, we all know that's nonsense, and incurs a _substantial_ downtime. Also, what's "non-commercial"? If I track my business todos (I'm a freelancer) – is that "a transaction of economic value"? I don't even care what the answer is – even the _requirement_ to think about this is a show-stopper in some way. Also, your legal team might even win even if you're in the wrong. 

I would also happily pay for that – a realistic price, though. As it is now, I might stop using the software once and for all, because _you simply never know what happens to non-open software_. And then my data is locked, and I am stuck with the migration.

Bottom line – while commercially understandable, that license is sh*t for users who are concerned about data longevity and trust. (sorry to be so blunt ...).

**Comment by flypenguin** [comment]
PS: I begin to really, really like the software. That's why I'm writing this – I don't want to freeload, I feel this might be the evolution of knowledge management I was looking for for my personal and professional needs. And this is creating a _big_ "unease", because you also didnt communicate that change ... very well.

Again: BIG FAN here. That's why.

**Comment by louwers** [comment]
Why do you link to a video "What is Open Source Software" from your "open code" page?

<img width="1056" alt="image" src="https://github.com/user-attachments/assets/358169ed-8370-4f4c-9d2a-8f691b4d4184">

**Comment by fuksman** [comment]
Hey @flypenguin, thanks for your messages.

I understand your concerns, and an option to change the license is still open for us, and we really wish to. Our choice of the source available license is dictated solely by the necessity to protect the growing product from copycats. Our infrastructure layer and data protocol are completely open-source, and only clients are source available. As a private user, you don't have any restrictions imposed by the license; basically, the only thing which the license forbids is selling access to Anytype without additional approval.

Regarding the risk that the license can somehow make you stuck with unrecoverable data, it is not possible. This is because the data format itself is licensed under MIT ([any-block](https://github.com/anyproto/any-block)), as well as the infrastructure layer ([any-sync](https://github.com/anyproto/any-sync)). And, just to repeat myself: you can build the client from sources for personal usage (work-related note-taking is personal use, only selling access to Anytype is a commercial use).

We really tried to communicate the change. Looks like we should work better on this, but here are our previous explanations:
- Our Open Philosophy: https://blog.anytype.io/our-open-philosophy/
- license-related FAQ https://legal.any.coop/

In short: we don't have evil in mind, our core values are that users should never lose access to their data, and we as a company shouldn't be able to restrict users from accessing their data; the only restriction we introduced in the license is an attempt to prevent bad actors from copying everything our team and contributors invested in before we find a way for sustainable development.

**Comment by flypenguin** [comment]
I fully and truly believe you that you – today – want all of that. But please tell me how it changes those two things in any way:

* if "anything" kills the client, an open protocol is worth absolutely _nothing_. Imagine MS pulls Word, and says "hey, it's an open data format – Open XML". Sure, no downtime then, and no migration work. The client is _my access_ to the data – open format or not.
* Also, I'm a Freelancer. Feel free to issue a legally binding statement here whether managing my freelance TODO items is "business use" ... (I'm willing to bet I will get not a whisper of an answer here, because it would be legal suicide for you to do so ;).

Again – I fully believe your intentions. It just does not change anything, because "trust us, everything is fine and forever will be" ... well. :)

Still: Big Fan. Really WANT to use it. Nonetheless have a bad feeling.

**Comment by fuksman** [comment]
I will have a try :)

- If anything kills the client (which, I assume, some changes in the license that would forbid what's currently not forbidden), you download the previous version or build/fork the sources with the previous license to keep using the product or export your data. It would work the same way even if the license is MIT or AGPL.
- I think you meant "Non-Commercial Use". If so, we can really issue this, because it's not what we're aiming to protect from. If this will add assurance in your case, we can start working on the statement and have it published until we open source all the clients.

**Comment by flypenguin** [comment]
Wow :)) . Thanks for trying.

* Okay, good point. Still, you see, it required me to be able to download ... etc. (Yes, I'm just arguing now, you _do_ have a point 😉)
* I know. Thanks for the offer, and I would assume a clarification of what _exactly_ you want to disallow could be helpful – at least it would be for me, and I cannot be the only freelancer (or similar person) using anytype ... boldly assumed :) .

And if you still intend to fully open-source everything – please please please please do so. :)

And bad feeling got smaller :) – thanks again!
